libcloud/compute/drivers/libvirt_driver.py

```diff
@@ -257,13 +257,27 @@ def _get_ip_addresses_for_domain(self, domain):
             # Only Linux is supported atm
             return result
 
+        if '///' not in self._uri:
+            # Only local libvirtd is supported atm
+            return result
+
         mac_addresses = self._get_mac_addresses_for_domain(domain=domain)
 
-        cmd = ['arp', '-an']
-        child = subprocess.Popen(cmd, stdout=subprocess.PIPE,
-                                 stderr=subprocess.PIPE)
-        stdout, _ = child.communicate()
-        arp_table = self._parse_arp_table(arp_output=stdout)
+        arp_table = {}
+        try:
+            cmd = ['arp', '-an']
+            child = subprocess.Popen(cmd, stdout=subprocess.PIPE,
```

Comment:
Hi Tomaz,

I had a version calling a different function for parsing the arp table of
'ip neighbor' but it looked like code duplication to me. So I rolled it
into the existing one.

I'd be happy to refactor it out into 2 functions. And update the tests when
they get in.

thanks
Rene
On May 28, 2016 10:25 AM, "Tomaz Muraus" <notifications@github.com> wrote:

> In libcloud/compute/drivers/libvirt_driver.py
> <https://github.com/apache/libcloud/pull/788#discussion_r64991609>:
>
> >          mac_addresses = self._get_mac_addresses_for_domain(domain=domain)
> >
> > -        cmd = ['arp', '-an']
> > -        child = subprocess.Popen(cmd, stdout=subprocess.PIPE,
> > -                                 stderr=subprocess.PIPE)
> > -        stdout, _ = child.communicate()
> > -        arp_table = self._parse_arp_table(arp_output=stdout)
> > +        arp_table = {}
> > +        try:
> > +            cmd = ['arp', '-an']
> > +            child = subprocess.Popen(cmd, stdout=subprocess.PIPE,
>
> Minor improvement which would also make it easier to test separately would
> be to refactor this into two methods (e.g. _get_arp_table_using_arp_cmd
> and _get_arp_table_using_ip_cmd or something along those lines).
>
> Anyways, not a blocker.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/libcloud/pull/788/files/786a95ddbac28e832794957b84008e7275fb99d8#r64991609>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADECW3-30sCa2IwJ9o13kJ93tx6FcGSLks5qGHqJgaJpZM4Ifqrg>
> .
>